merlin/models/tf/utils/tf_utils.py

```diff
@@ -469,3 +469,29 @@ def list_col_to_ragged(col: Tuple[tf.Tensor, tf.Tensor]):
         row_lengths = tf.cast(row_lengths, tf.int32)
 
     return tf.RaggedTensor.from_row_lengths(values, row_lengths)
+
+
+def check_inputs_mask_compatible_shape(
```

Comment:
If the inputs and mask don’t have the same type, the function returns `False`. So we could maybe raise the exception inside the block that uses the util function, rather than modifying the function itself? Like [here](https://github.com/NVIDIA-Merlin/models/blob/95f48f61f5381ff583f9c062b3dad672bd271847/merlin/models/tf/transforms/sequence.py#L812
) in the `ReplaceMaskedembeddings` block.